### PR TITLE
pkg: tests: handle arch requirements at pkg level, remove useless board blacklist

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is using the recently added arch requirements in the dependency files at package level. This allows to remove the board blacklist used in package related tests/examples.

Only packages and their related tests/examples are affected by this PR.

### Testing procedure

A green Murdock should be ok.

### Issues/PRs references

Partial (but cleaner) alternative to #12577.

### CODE REVIEW

**Comment by Rotzbua** [comment]
`arch_32bit` is supported by esp8266 but is in some blacklists

File: pkg/wolfssl/Makefile.dep [hunk 1]
```diff
@@ -82,3 +82,6 @@ endif
 ifneq (,$(filter wolfcrypt_random,$(USEMODULE)))
   USEMODULE += random
 endif
+
+# wolfssl is only supported by 32 bit architectures
+FEATURES_REQUIRED += arch_32bit
```
Comment (miri64) [inline pkg/wolfssl/Makefile.dep]: @Rotzbua is right. This package needs a 

```
FEATURES_BLACKLIST += arch_esp8266
```

**Comment by kaspar030** [comment]
Output of all "make info-board-supported" should be compared.

File: pkg/wolfssl/Makefile.dep (hunk 1 above)
Comment (aabadie) [reply pkg/wolfssl/Makefile.dep]: The comment was only saying "only 32 bit architectures". Do we know why esp8266 boards were blacklisted ?

Comment (miri64) [reply pkg/wolfssl/Makefile.dep]: Sadly, no. Weirdly enough, in https://github.com/RIOT-OS/RIOT/pull/12454 the ESP8266 wasn't a problem. Maybe @danielinux, @maribu, @MichelRottleuthner, or @smlng could give some insights?

Comment (aabadie) [reply pkg/wolfssl/Makefile.dep]: and #10308 doesn't give more info unfortunately.

**Comment by aabadie** [comment]
Added a couple of commit for applying the same cleanup to the tinydtls package and the associated dtls-echo example.

File: pkg/wolfssl/Makefile.dep (hunk 1 above)
Comment (MichelRottleuthner) [reply pkg/wolfssl/Makefile.dep]: Maybe it was just blacklisted by accident or @danielinux had problems with it? I don't remember discussing any specific problems of wolfssl on the esp but also I didn't test on that platform.

**Comment by aabadie** [comment]
Added commits for lua and jerryscript packages.

File: examples/javascript/Makefile [hunk 2]
```diff
@@ -7,11 +7,10 @@ BOARD ?= native
...
 RIOTBASE ?= $(CURDIR)/../..
 
-BOARD_BLACKLIST := arduino-duemilanove arduino-leonardo arduino-mega2560 \
-                   arduino-nano arduino-uno atmega328p chronos \
-                   msb-430 msb-430h telosb waspmote-pro wsn430-v1_3b \
-                   wsn430-v1_4 z1 pic32-wifire pic32-clicker \
-                   mega-xplained microduino-corerf
+BOARD_BLACKLIST += \
+    pic32-wifire   \
+    pic32-clicker  \
```
Comment (miri64) [inline examples/javascript/Makefile]: `FEATURES_BLACKLIST += arch_mips32r2` to `pkg/jerryscript`?

File: examples/lua_REPL/Makefile [hunk 3]
```diff
@@ -6,11 +6,12 @@ BOARD ?= native
...
-BOARD_BLACKLIST := arduino-duemilanove arduino-leonardo \
-                   arduino-mega2560 arduino-nano arduino-uno atmega328p \
-                   chronos hifive1 hifive1b mega-xplained microduino-corerf \
-                   msb-430 msb-430h pic32-clicker pic32-wifire telosb \
-                   waspmote-pro wsn430-v1_3b wsn430-v1_4 z1
+BOARD_BLACKLIST +=  \
+    hifive1         \
+    hifive1b        \
+    pic32-wifire    \
+    pic32-clicker   \
```
Comment (miri64) [inline examples/lua_REPL/Makefile]: Here and below: `FEATURES_BLACKLIST += arch_mips32r2 arch_riscv` to `pkg/lua`?

Comment (aabadie) [reply examples/lua_REPL/Makefile]: Done for MIPS. For RISCV, I don't see any reason why the boards is blacklisted. Will see if I can find some info in related PRs.

Comment (aabadie) [reply examples/lua_REPL/Makefile]: For RISCV, the problem is missing functions in the newlib nano provided by the toolchain. Not sure if simply blacklisting them is the way to go but I did it anyway.

**Comment by aabadie** [comment]
I reran the script from above with the last version of this PR and got the same results.

File: examples/lua_REPL/Makefile (hunk 3 above)
Comment (miri64) [reply examples/lua_REPL/Makefile]: What alternative would you propose?

**Comment by aabadie** [comment]
@miri64, @kaspar030, I think we are good here. Do you still have concerns ?

**Comment by miri64** [comment]
My comments were addressed.

**Comment by benpicco** [comment]
> My comments were addressed.

yet no ACK?

**Comment by aabadie** [comment]
Thank you for the reviews!

**Comment by miri64** [comment]
> > My comments were addressed.
> 
> yet no ACK?

Sorry, didn't have the time to have a look beyond confirming that my comments were addressed last week.

**Pull request merged**